### PR TITLE
Clean up duplicate code and remove import cycle.

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In #86282, we deprecated all metrics under endpoint `/metrics/resource/v1alpha1` which locate at [pkg/kubelet/apis/resourcemetrics/v1alpha1/config.go](https://github.com/kubernetes/kubernetes/blob/b6b494b4484b51df8dc6b692fab234573da30ab4/pkg/kubelet/apis/resourcemetrics/v1alpha1/config.go).

It's hard to do a unit test for these deprecated metrics because of the import cycle.
Let's say :
- the package `pkg/kubelet/server` is `A`.
- the package `pkg/kubelet/server/stats` is `B`.
- the package `pkg/kubelet/apis/resourcemetrics/v1alpha1` is `C`.

`A` has both dependencies of `B` and `C` at [here](https://github.com/kubernetes/kubernetes/blob/6e35a13815adf221bc2b65eadcdb326de2283521/pkg/kubelet/server/server.go#L353). 
`C` has dependencies of `B` because of [stats.ResourceMetricsConfig](https://github.com/kubernetes/kubernetes/blob/6e35a13815adf221bc2b65eadcdb326de2283521/pkg/kubelet/apis/resourcemetrics/v1alpha1/config.go).

Given we want to test the collector in package `B`, by a call like  [NewPrometheusResourceMetricCollector(s.resourceAnalyzer, v1alpha1.Config())](https://github.com/kubernetes/kubernetes/blob/6e35a13815adf221bc2b65eadcdb326de2283521/pkg/kubelet/server/server.go#L353), that will involve `C`. The import cycle happens...

This PR intends to eliminate the import cycle and revise the ugly unit tests. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/priority backlog

### CODE REVIEW

File: pkg/kubelet/server/stats/prometheus_resource_metrics.go [hunk 1]
```diff
@@ -24,6 +24,77 @@ import (
 	stats "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
 )
 
+// This file contains a series of deprecated metrics which we emit them by endpoint `/metrics/resource/v1alpha1`.
+// These metrics have been adapted to new endpoint `/metrics/resource` as well as new `Desc`s.
+// In general, we don't need to maintain these deprecated metrics any more.
+// TODO(RainbowMango): Remove this file in release 1.20.0+.
```
Comment (mattjmcnaughton) [inline pkg/kubelet/server/stats/prometheus_resource_metrics.go]: Just to confirm... we want to remove the entire `prometheus_resource_metrics.go` file in 1.20?

File: pkg/kubelet/apis/resourcemetrics/v1alpha1/config.go [hunk 2]
```diff
@@ -16,110 +16,8 @@ limitations under the License.
...
-
-// This file contains a series of deprecated metrics which we emit them by endpoint `/metrics/resource/v1alpha1`.
-// These metrics have been adapted to new endpoint `/metrics/resource` as well as new `Desc`s.
-// In general, we don't need to maintain these deprecated metrics any more.
-// TODO(RainbowMango): Remove this file in release 1.20.0+.
+// TODO(RainbowMango): We don't need to maintain this package anymore.
+//  This package will be remove in release 1.20.0+. More details please refer to https://github.com/kubernetes/kubernetes/pull/86282.
 
 // Version is the string representation of the version of this configuration
 const Version = "v1alpha1"
```
Comment (mattjmcnaughton) [inline pkg/kubelet/apis/resourcemetrics/v1alpha1/config.go]: Out of curiosity, why are we keeping this file with just the `Version` constant defined?

Do we not want to keep the version definition with the actual `NewDesc` descriptions `pkg/kubelet/server/stats/prometheus_resource_metrics.go`?

**Review by mattjmcnaughton — COMMENTED** [review]
A couple questions:

1. Do you mind updating either the PR description or the commit message with a description of the import cycle (i.e. what packages were originally involved in the cycle and how did you break it)? 
2. Given we plan to delete all of this code in 1.20, what is the advantage of undertaking this refactor now? In other words, does this refactor only impact code that we already plan to delete in 1.20?

Thanks :)

**Comment by dashpole** [comment]
/approve
thanks for opening this.  This lgtm after mattjmcnaughton@'s question about the verison is resolved.

File: pkg/kubelet/server/stats/prometheus_resource_metrics.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/kubelet/server/stats/prometheus_resource_metrics.go]: 100% right. Please see comments above
`These metrics have been adapted to new endpoint `/metrics/resource` as well as new `Desc`s.`.

File: pkg/kubelet/apis/resourcemetrics/v1alpha1/config.go (hunk 2 above)
Comment (RainbowMango) [reply pkg/kubelet/apis/resourcemetrics/v1alpha1/config.go]: It's no big deal in my opinion. Given the constant also be referred by other components, such as [E2E](https://github.com/kubernetes/kubernetes/blob/6e35a13815adf221bc2b65eadcdb326de2283521/test/e2e_node/resource_metrics_test.go#L114), so I'd prefer to clean it at release 1.20.

File: pkg/kubelet/server/stats/prometheus_resource_metrics.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/kubelet/server/stats/prometheus_resource_metrics.go]: And these metrics have almost a copy at [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/collectors/resource_metrics.go).

**Comment by RainbowMango** [comment]
@mattjmcnaughton 

> Do you mind updating either the PR description...

Updated.

> In other words, does this refactor only impact code that we already plan to delete in 1.20?

Yes, this change only impacts the deprecated metrics. And the behavior of these metrics hasn't changed.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Ok, I'm comfortable with this change. Given this code will all be deleted in 1.20, if I thought there could be any downsides to this refactor, I might advise against it. But, I don't see any reason for concern, so doing this refactor means we can enjoy cleaner metrics code from 1.18-1.20.

Thanks for updating the pr description!

**Comment by RainbowMango** [comment]
Kindly ping @derekwaynecarr .

/assign @yujuhong 
(if you have the review bandwidth for this, it would be greatly appreciated)

**Pull request merged**